djangocms_pageadmin/admin.py

```diff
@@ -284,14 +318,56 @@ def duplicate_view(self, request, object_id):
             request, "djangocms_pageadmin/admin/duplicate_confirmation.html", context
         )
 
+    @require_POST
+    @transaction.atomic
+    def set_home_view(self, request, object_id):
+        page_content = self.get_object(request, object_id=unquote(object_id))
+
+        if not self.has_change_permission(request, page_content):
+            raise PermissionDenied("You do not have permission to set 'home'.")
```

Comment:
I think this should check permissions on Page object, and not PageContent.

Reply:
OK, move the code a bit as well...